posts/a-guinea-pig-in-the-cloud.md

```diff
@@ -0,0 +1,389 @@
+
+![Integration mode](https://github.com/bpinel/blog-usa/master/images/2019/02/ember-aws-ehipster-int.png)
+
+*   The **production mode** just pushes the web resources to a S3 bucket exposed through CloudFront. The application can then be accessed all over the world!
+
+ ![Production mode](https://github.com/bpinel/blog-usa/master/images/2019/02/ember-aws-ehipster-prod.png)
+
+In a future post, I will show a way to also integrate Cognito to set up some authentication and authorization directly on the Cloud.
+
+# Let's demonstrate!
+
+Let's now set up a complete example and a full workflow going from development to 'production'. The term 'production' is surrounded by quotes because the implementation that I have done is just a Proof of Concept. A production-ready implementation would deserve much more reflection and work… 
+
+## Setting up the development environment
+
+The story begins by starting a standard Ember.js application with Ember CLI. To proceed, you should already have [Node.js](https://nodejs.org) and [Ember CLI](https://ember-cli.com/) installed on your computer. Then you just type the usual:
+
+```shell
+ember new guineapig_inthecloud --no-welcome
+```
+
+The _no_welcome_ option is just there to get rid of the welcome screen set by default for an Ember application. 
+
+To complete this installation, you just need to install my add-on _ember-aws-ehipster_ by typing:
+
+```shell
+cd guineapig_inthecloud
+ember install ember-aws-ehipster
+```
+
+This add-on will install several other useful add-ons (listed in the page [https://gitlab.ippon.fr/bpinel/ember-aws-ehipster/wikis/ember-included-add-ons](https://gitlab.ippon.fr/bpinel/ember-aws-ehipster/wikis/ember-included-add-ons)) and add an Ember blueprint for automatically generating model objects, mock data and administration pages from a descriptive file. This file is simply the one you get when designing a model through [JDL-Studio](https://start.jhipster.tech/jdl-studio/) from JHipster. Here is the one we will use in this example : 
+
+```
+entity User {
+ login String required minlength(4),
+ passwordHash String required minlength(8),
+ firstName String required minlength(3) maxlength(50),
+ lastName String,
+ email String,
+ imageUrl String,
+ activated Boolean,
+ createdBy String,
+ createdDate LocalDate,
+ lastModifiedBy String,
+ lastModifiedDate Instant 
+}
+
+entity Authority {
+ name String required maxlength(30)
+}
+
+entity Todo {
+  title String required,
+  completed Boolean
+}
+
+relationship ManyToMany {
+	User{user(login)} to Authority{authority(name)}
+}
+
+relationship OneToMany {
+ 	User{user(login)} to Todo{todo}
+}
+```
+
+Note that **Authority** and **User** objects are just taken from the JHipster model and can be represented by the following diagram:
+![JDL model](https://github.com/bpinel/blog-usa/blob/master/images/2019/02/jdl-todo.png)
```

Comment:
✅ FYI, this image, and all the subsequent images, work correctly.

Reply:
I miss the /blob part in the previous URL... Including in the one referenced from the metadata (tag 'image:'). Can you correct it ?